docker/storage/elasticsearch6/Dockerfile

```diff
@@ -12,41 +12,39 @@
 # the License.
 #
 
-FROM alpine:3.12
+FROM openzipkin/zipkin-builder as install
 
-ENV ELASTICSEARCH_VERSION 6.8.11
+ENV ELASTICSEARCH_VERSION 6.8.12
 
-RUN apk add --update curl
+WORKDIR /install
 
-WORKDIR /elasticsearch
+# We don't download bin scripts as we customize for reasons including BusyBox problems
+RUN curl -sSL https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-$ELASTICSEARCH_VERSION.tar.gz| tar xz \
+    --wildcards --strip=1 --exclude=*/bin
 
-RUN curl -sSL https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-$ELASTICSEARCH_VERSION.tar.gz| tar xz && \
-    mv elasticsearch-$ELASTICSEARCH_VERSION/* /elasticsearch/ && \
-    # remove binaries and non shell scripts
-    rm -f /elasticsearch/bin/elasticsearch-sql* /elasticsearch/bin/*cmd /elasticsearch/bin/*exe && \
-    # Remove bash as our images doesn't have it, and it isn't required
-    ls -d /elasticsearch/bin/* | grep -v x-pack | xargs sed -i 's~#!/bin/bash~#!/bin/sh~g' && \
-    chmod +x /elasticsearch/bin/*
-
-COPY docker/storage/elasticsearch6/config /elasticsearch/config
+COPY docker/storage/elasticsearch6/config /install/config
 
 FROM openzipkin/jre-full:14.0.2-14.29.23
 LABEL MAINTAINER Zipkin "https://zipkin.io/"
 
-# https://github.com/elastic/elasticsearch/pull/31003 was closed won't fix
-ENV ES_TMPDIR /tmp
-
-RUN adduser -g '' -h /elasticsearch -D elasticsearch
+# Add HEALTHCHECK and ENTRYPOINT scripts into the default search path
+COPY docker/storage/elasticsearch7/docker-bin/* /usr/local/bin/
```

Comment:
intentionally using the same scripts. didn't extract a common "elasticsearch" directory over this, as it would likely be more distracting.